src/renderer/services/neo.js

```diff
@@ -573,6 +704,98 @@ export default {
           pushSystemAssetHolding(GAS_ASSET_ID, 'GAS', 0);
         }
 
+        const pushNewHolding = (assetId, name, symbol, balance, decimals,
+          canPull, isETHToken, abi, availableSupply) => {
+          const systemAssetHolding = {
+            assetId,
+            balance: new BigNumber(balance),
+            symbol,
+            name,
+            contractBalance: new BigNumber(0),
+            totalBalance: new BigNumber(0),
+            decimals,
+            isUserAsset: true,
+            canPull,
+            isETHToken,
+            abi,
+            availableSupply,
+          };
+          holdings.push(systemAssetHolding);
+        };
+
+        // ETH System Holding
+
+        const ethAddress = currentWallet.ethAddress;
+        let balEthTemp;
+        let balEth;
+
+        try {
+          balEthTemp = await web3.eth.getBalance(ethAddress);
+          balEth = toBigNumber(balEthTemp).dividedBy(10 ** 18);
+        } catch (e) {
+          balEthTemp = 0;
+          balEth = 0;
```

Comment:
It's better to keep the last good value received than to make the balance jump to 0 if it can't be retrieved.